prometheus/promhttp/http.go

```diff
@@ -165,19 +186,13 @@ func HandlerForTransactional(reg prometheus.TransactionalGatherer, opts HandlerO
 		} else {
 			contentType = expfmt.Negotiate(req.Header)
 		}
-		header := rsp.Header()
-		header.Set(contentTypeHeader, string(contentType))
-
-		w := io.Writer(rsp)
-		if !opts.DisableCompression && gzipAccepted(req.Header) {
-			header.Set(contentEncodingHeader, "gzip")
-			gz := gzipPool.Get().(*gzip.Writer)
-			defer gzipPool.Put(gz)
-
-			gz.Reset(w)
-			defer gz.Close()
+		rsp.Header().Set(contentTypeHeader, string(contentType))
 
-			w = gz
+		w, err := GetWriter(req, rsp, opts.DisableCompression, opts.OfferedCompressions)
+		if err != nil {
+			if opts.ErrorLog != nil {
+				opts.ErrorLog.Println("error getting writer", err)
+			}
```

Comment:
It would be nice to explictly do `w = io.Writer(rsp)`  and not rely on GetWriter implementation to do something NOT Go idiomatic (give valid response despite err != nil), WDYT?

Reply:
Good point, the function doesn't return a writer anymore in case there's an error.